tools/swift-inspect/Sources/SwiftInspectHeapWalk/SwiftInspectHeapWalk.cpp

```diff
@@ -0,0 +1,184 @@
+#if defined(_WIN32)
+
+#pragma comment(lib, "swiftCore.lib")
+
+#include <windows.h>
+#include <stdio.h>
+
+namespace {
+struct HeapEntry {
+  uintptr_t address;
+  uintptr_t size;
+};
+} // anonymous namespace
+
+#define BUF_SIZE 512
+#define BUF_NUM_ENTRIES (BUF_SIZE / sizeof(HeapEntry))
+#define SHARED_MEM_NAME "Local\\SwiftInspectFileMapping"
+#define READ_EVENT_NAME "Local\\SwiftInspectReadEvent"
+#define WRITE_EVENT_NAME "Local\\SwiftInspectWriteEvent"
+#define WAIT_TIMEOUT_MS 30000
+
+static int heapWalk() {
+  HANDLE hMapFile = OpenFileMappingA(FILE_MAP_ALL_ACCESS, false,
+                                     SHARED_MEM_NAME);
+  if (hMapFile == NULL) {
+    printf("OpenFileMapping failed\n");
+    return 1;
+  }
+  HeapEntry *buf = (HeapEntry *)
```

Comment:
Since the file is marked as C++ rather than C, please use `reinterpret_cast` rather than the C-style cast.